### PR TITLE
Improve release process

### PR DESCRIPTION
**Pull request by bkchr**
This pull request introduces the `changelog-processor.py`. A small script for getting some information out of the `CHANGELOG.md` file for improving the release process. The script supports to get the latest version in the `CHANGELOG.md` file, aka the version top of the file. (This could also be `Unreleased`) The other command is for printing if there should be done a release or not. This is decided based on the latest version and if there exists a tag for it.

Closes: https://github.com/polkadot-fellows/runtimes/issues/40

### CODE REVIEW

**Review by xlc — APPROVED** [review]

File: .github/changelog-processor.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    for line in lines:
+        if not line.startswith("## ["):
+            continue
+
+        version = line.strip().removeprefix("## [").split("]")[0]
+        break
+
+    if args.print_latest_version:
+        print(version, end = "")
+        sys.exit(1)
```
Comment (ggwpez) [inline .github/changelog-processor.py:41]: Why does it error in this case?

File: .github/changelog-processor.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import os
+import sys
+import argparse
+
+parser = argparse.ArgumentParser(description="Process the CHANGELOG.md")
+parser.add_argument(
+    "changelog",
+    metavar="CHANGELOG",
+    help="Path to the CHANGELOG.md file"
```
Comment (ggwpez) [inline .github/changelog-processor.py:11]: You can make this optional:

```suggestion
    help="Path to the CHANGELOG.md file",
    default="CHANGELOG.md",
    nargs='?'
```

File: .github/changelog-processor.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+)
+parser.add_argument(
+    "--should-release",
+    dest="should_release",
+    help="Should a release be made? Prints `1` or `0`.",
+    action="store_true"
+)
+
+args = parser.parse_args()
+
```
Comment (ggwpez) [inline .github/changelog-processor.py:27]: Could sanity check that only one of the two flags are set.

**Review by ggwpez — DISMISSED** [review]

**Review by acatangiu — APPROVED** [review]

File: .github/changelog-processor.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        stream = os.popen("git tag -l v" + version)
+        output = stream.read()
+
+        # Empty output means that the tag doesn't exist and that we should release.
+        if output.strip() == "":
+            print("1", end = "")
+        else:
+            print("0", end = "")
+
+        sys.exit(1)
```
Comment (ordian) [inline .github/changelog-processor.py:61]: also in this case?

File: .github/changelog-processor.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    for line in lines:
+        if not line.startswith("## ["):
+            continue
+
+        version = line.strip().removeprefix("## [").split("]")[0]
+        break
+
+    if args.print_latest_version:
+        print(version, end = "")
+        sys.exit(1)
```
Comment (bkchr) [inline .github/changelog-processor.py:43]:
```suggestion
        sys.exit(0)
```

File: .github/changelog-processor.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        stream = os.popen("git tag -l v" + version)
+        output = stream.read()
+
+        # Empty output means that the tag doesn't exist and that we should release.
+        if output.strip() == "":
+            print("1", end = "")
+        else:
+            print("0", end = "")
+
+        sys.exit(1)
```
Comment (bkchr) [inline .github/changelog-processor.py:61]:
```suggestion
        sys.exit(0)
```

File: .github/changelog-processor.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        version = line.strip().removeprefix("## [").split("]")[0]
+        break
+
+    if args.print_latest_version:
+        print(version, end = "")
+        sys.exit(1)
+    elif args.should_release:
+        if version.lower() == "unreleased":
+            print("0", end = "")
+            sys.exit(1)
```
Comment (bkchr) [inline .github/changelog-processor.py:47]:
```suggestion
            sys.exit(-1)
```

**Review by ordian — APPROVED** [review]
looks good modulo unresolved comments

File: .github/changelog-processor.py (hunk 1 above)
Comment (bkchr) [reply .github/changelog-processor.py:43]: My brain wasn't working :P I had forgotten that `0` is "success"

**Review by ggwpez — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

**Pull request merged**